app/assets/javascripts/discourse/tests/unit/models/wizard-field-test.js

```diff
@@ -1,44 +1,38 @@
-import { getOwner } from "discourse-common/lib/get-owner";
+import { Field } from "wizard/models/wizard";
 import { setupTest } from "ember-qunit";
 import { module, test } from "qunit";
 
 module("Unit | Model | Wizard | wizard-field", function (hooks) {
   setupTest(hooks);
 
   test("basic state", function (assert) {
-    const store = getOwner(this).lookup("service:store");
-    const field = store.createRecord("wizard-field", { type: "text" });
+    const field = new Field({ type: "text" });
```

Comment:
In general we've been trying to move away from `new Blah()` and towards `store.createRecord` so that the instance's owner is set correctly, and service injections work as expected.

In the wizard case, I guess it doesn't matter too much because the models have no need for an owner. But ideally we would have a consistent pattern which can be applied everywhere. What do you think? Is `store.createRecord` a good idea? Or should be aiming to come up with an alternative strategy which keeps service injections working?

Reply:
I this case, I think it's probably fine, because I have refactored the models to be plain JS classes. If an owner is needed for injection, it should be taken in the constructor and set in the constructor body:

```js
class MyClass {
  constructor(owner, ...) {
    setOwner(this, owner);
  }
}
```

Using `store.createRecord` "works", but it kind of only masks the problem and introduces some other ones. It relies on the object being a subclass of `Ember.Object` so that the "protocol" of `.create({ [OWNER]: owner, ... })` works, but that isn't a general purpose JavaScript pattern (Glimmer Components aren't subclass of `Ember.Object` for instance), it also relies on the string resolution of "wizard-field" into the actual model class, which, in general, and especially in this case, is counterproductive to moving this part of the app to being "static" for Embroider